indra/newview/llviewerobject.cpp

```diff
@@ -3466,6 +3555,52 @@ BOOL LLViewerObject::loadTaskInvFile(const std::string& filename)
     return TRUE;
 }
 
+BOOL LLViewerObject::loadTaskInvLLSD(const LLSD& inv_result)
+{
+    if (inv_result.has("contents"))
+    {
+        if(mInventory)
+        {
+            mInventory->clear(); // will deref and delete it
+        }
+        else
+        {
+            mInventory = new LLInventoryObject::object_list_t;
+        }
+
+        // Synthesize the "Contents" category, the viewer expects it, but it isn't sent.
+        LLPointer<LLInventoryObject> inv = new LLInventoryObject(mID, LLUUID::null, LLAssetType::AT_CATEGORY, "Contents");
+        mInventory->push_front(inv);
+
+        const LLSD& inventory = inv_result["contents"];
+        auto iter = inventory.beginArray();
+        auto end = inventory.endArray();
+        for(; iter != end; ++iter)
```

Comment:
```
for (const auto& inv_entry : llsd::inArray(inventory))
{
    if (inv_entry.has("item_id"))
```

Reply:
Aha, was hoping there was an API like that, thanks!